cpp/src/io/json/reader_impl.cu

```diff
@@ -600,9 +601,22 @@ table_with_metadata reader::impl::convert_data_to_table(rmm::cuda_stream_view st
   stream.synchronize();
 
   // postprocess columns
-  auto target = make_strings_column(
-    std::vector<char>{'\\', '"', '\\', '\\', '\\', 't', '\\', 'r', '\\', 'b'}, {0, 2, 4, 6, 8, 10});
-  auto repl = make_strings_column({'"', '\\', '\t', '\r', '\b'}, {0, 1, 2, 3, 4, 5});
+  auto target_chars   = std::vector<char>{'\\', '"', '\\', '\\', '\\', 't', '\\', 'r', '\\', 'b'};
+  auto target_offsets = std::vector<size_type>{0, 2, 4, 6, 8, 10};
+
+  auto repl_chars   = std::vector<char>{'"', '\\', '\t', '\r', '\b'};
+  auto repl_offsets = std::vector<size_type>{0, 1, 2, 3, 4, 5};
+
+  auto target = make_strings_column(cudf::detail::make_device_uvector_async(target_chars, stream),
+                                    cudf::detail::make_device_uvector_async(target_offsets, stream),
+                                    {},
+                                    0,
+                                    stream);
+  auto repl   = make_strings_column(cudf::detail::make_device_uvector_async(repl_chars, stream),
+                                  cudf::detail::make_device_uvector_async(repl_offsets, stream),
+                                  {},
+                                  0,
+                                  stream);
 
   thrust::host_vector<cudf::size_type> h_valid_counts = d_valid_counts;
```

Comment:
This is doing a D2H copy w/o a stream.

Reply:
There are many other bugs that this PR does not fix as well! :)

Reply:
This function also uses a log of device_vectors. Replacing them is non-trivial (unlike some of the strings changes I made), so I prefer to tackle that in a separate PR.